### ISSUE TITLE
ws: change response format

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#19774]
*Part of epic https://github.com/MaterializeInc/materialize/issues/19771.*

From https://github.com/MaterializeInc/materialize/issues/18224#issuecomment-1562288760:

> Say, the way the response format always uses `payload` isn't very JS/TS-y:
> 
> ```
> {"type":"Rows","payload":["?column?","?column?"]}
> {"type":"Row","payload":["1","2"]}
> {"type":"CommandComplete","payload":"SELECT 1"}
> {"type":"Rows","payload":["column1"]}
> {"type":"Row","payload":["4"]}
> {"type":"Row","payload":["5"]}
> {"type":"CommandComplete","payload":"SELECT 2"}
> {"type":"ReadyForQuery","payload":"I"}
> ```
> 
> It's more common to use descriptive field names like:
> 
> ```
> {"type":"Rows","columns":["?column?","?column?"], "statementType": "SELECT"}
> {"type":"Row","row":["1","2"]}
> {"type":"CommandComplete","tag":"SELECT 1"}
> ```
> 
> Serde calls this the "internally tagged" enum style.

cc @mjibson

**Comment by mjibson** [comment]
Yup. I don't remember why I did it this way. Might have been due to unfamiliarity with serde? But yes, we should do this.
